.github/workflows/e2e_test_reusable.yml

```diff
@@ -0,0 +1,92 @@
+on:
+  workflow_call:
+
+jobs:
+  prepare-matrix:
+    name: Prepare Matrix
+    runs-on: ubuntu-latest
+    # Define the outputs that this job will produce
+    outputs:
+        files: ${{ steps.set-files.outputs.FILES }}
+    steps:
+      # Checkout the code
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+
+      # Set the files to run
+      - id: set-files
+        run: |
+          FILE_LIST=$(ls frontend/cypress/e2e/*.cy.js | jq -R -s -c 'split("\n")[:-1]')
```

Comment:
| Part | Description |
|------|-------------|
| `ls frontend/cypress/e2e/*.cy.js` | This part lists all files with the `.cy.js` extension in the `frontend/cypress/e2e/` directory. |
| `jq -R -s -c 'split("\n")[:-1]'` | This part of the command processes the output of the `ls` command using `jq`. It reads raw input (`-R`), reads the entire input stream into memory as a single string (`-s`), and outputs the result as a single-line JSON object (`-c`). The `'split("\n")[:-1]'` part splits the input string by newline characters and selects all elements except the last one. This is used to remove the trailing empty string that `jq` adds when splitting. |